### PR TITLE
Add the azure openai finetune connector

### PR DESCRIPTION
**Pull request by isuruRuhu**
# Description

Add a connector for azure openAI fine-tuning, files and model APIs

Fixes https://github.com/wso2-enterprise/choreo/issues/19293

One line release note: 
- Add a connector for azure openAI fine-tuning, files and model APIs

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Ballerina Version: 2201.4.1
* Operating System: macOS Ventura (version 13.0.1)
* Java SDK: openjdk version "17.0.6"

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [x] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [ ] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/azure.openai.finetunes/Module.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+## Overview
+This is a generated connector from [Azure OpenAI Files API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/files/),
+[Azure OpenAI Fine Tunes API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/fine-tunes/) and
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Module.md:3]:
```suggestion
[Azure OpenAI Fine Tunes API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/fine-tunes/), and
```

File: openapi/azure.openai.finetunes/Module.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+[Azure OpenAI Models API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/models/) OpenAPI specification.
+
+The Azure OpenAI Files API give access to data files related operations such as delete, get, get content, import, list and upload. The Azure OpenAI Fine Tunes API give access to fine tuning of base models and related operations such as create, cancel, delete, get, get event and list. The Azure OpenAI  Models API give access to deteails about the existing models.
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Module.md:6]:
```suggestion
The Azure OpenAI Files API gives access to data files related operations such as delete, get, get content, import, list, and upload. The Azure OpenAI Fine Tunes API gives access to fine-tuning of base models and related operations such as create, cancel, delete, get, get event, and list. The Azure OpenAI  Models API gives access to details about the existing models.
```

File: openapi/azure.openai.finetunes/Module.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Prerequisites
+- Create an [Azure](https://azure.microsoft.com/en-us/features/azure-portal/) account.
+- Create an [Azure OpenAI resource](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource).
+- Obtain the tokens. Refer [this](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#authentication) guide to learn how to generate and use tokens.
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Module.md:11]:
```suggestion
- Obtain the tokens. Refer to [Azure OpenAI Authentication](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#authentication) guide to learn how to generate and use tokens.
```

File: openapi/azure.openai.finetunes/Module.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+## Quickstart
+
+To use the Azure OpenAI Fine Tunes connector in your Ballerina application, update the .bal file as follows:
+
+### Step 1: Import connector
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Module.md:17]:
```suggestion
### Step 1: Import the connector
```

File: openapi/azure.openai.finetunes/Module.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+Import the `ballerinax/azure.openai.finetunes` module into the Ballerina project.
+
+```ballerina
+import ballerinax/azure.finetunes;
+```
+
+### Step 2: Create a new connector instance
+
+Create and initialize a `finetunes:Client` with the obtained `apiKey` and a `serviceUrl` from the azure OpenAI resource
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Module.md:26]:
```suggestion
Create and initialize a `finetunes:Client` with the obtained `apiKey` and a `serviceUrl` from the Azure OpenAI resource
```

File: openapi/azure.openai.finetunes/Module.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    serviceUrl = serviceUrl
+);
+```
+
+### Step 3: Invoke connector operation
+1. Now you can use the operations available within the connector. 
+
+>**Note:** that they are in the form of remote operations.
+
+Following is an example on uploading a file and finetuning curie base model:
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Module.md:40]:
```suggestion
Following is an example of uploading a file and finetuning curie base model:
```

File: openapi/azure.openai.finetunes/Module.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+Uploading the data file from local machine.
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Module.md:42]:
```suggestion
Uploading the data file from the local machine.
```

File: openapi/azure.openai.finetunes/Package.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+Connects to [Azure OpenAI Files API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/files/),
+[Azure OpenAI Fine Tunes API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/fine-tunes/) and
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Package.md:2]:
```suggestion
[Azure OpenAI Fine Tunes API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/fine-tunes/), and
```

File: openapi/azure.openai.finetunes/Package.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+[Azure OpenAI Models API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/azureopenaistable/models/) from Ballerina.
+
+## Package overview
+The `azure.openai.finetunes` is a [Ballerina](https://ballerina.io/) connector for Azure OpenAI Service Files API, Azure OpenAI Fine Tunes API and Azure OpenAI Models API.
```
Comment (anupama-pathirage) [inline openapi/azure.openai.finetunes/Package.md:6]:
```suggestion
The `azure.openai.finetunes` is a [Ballerina](https://ballerina.io/) connector for Azure OpenAI Service Files API, Azure OpenAI Fine Tunes API, and Azure OpenAI Models API.
```

File: openapi/azure.openai.finetunes/Module.md (hunk 1 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Module.md:3]: Modified module.md file according to the comment

File: openapi/azure.openai.finetunes/Module.md (hunk 2 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Module.md:6]: Modified module.md file according to the comment

File: openapi/azure.openai.finetunes/Module.md (hunk 3 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Module.md:11]: Modified module.md file according to the comment

File: openapi/azure.openai.finetunes/Module.md (hunk 4 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Module.md:17]: Modified module.md file according to the comment

File: openapi/azure.openai.finetunes/Module.md (hunk 5 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Module.md:26]: Modified module.md file according to the comment

File: openapi/azure.openai.finetunes/Module.md (hunk 6 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Module.md:40]: Modified module.md file according to the comment

Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Module.md:40]: Modified module.md file according to the comment

File: openapi/azure.openai.finetunes/Module.md (hunk 7 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Module.md:42]: Modified module.md file according to the comment

File: openapi/azure.openai.finetunes/Package.md (hunk 8 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Package.md:2]: Modified package.md file according to the comment

File: openapi/azure.openai.finetunes/Package.md (hunk 9 above)
Comment (isuruRuhu) [reply openapi/azure.openai.finetunes/Package.md:6]: Modified package.md file according to the comment

**Review by anupama-pathirage — APPROVED** [review]

File: openapi/azure.openai.finetunes/Module.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+## Quickstart
+
+To use the Azure OpenAI Fine Tunes connector in your Ballerina application, update the .bal file as follows:
+
+### Step 1: Import the connector
+Import the `ballerinax/azure.openai.finetunes` module into the Ballerina project.
+
+```ballerina
+import ballerinax/azure.finetunes;
```
Comment (NipunaRanasinghe) [inline openapi/azure.openai.finetunes/Module.md:21]:
```suggestion
import ballerinax/azure.openai.finetunes;
```

**Review by abeykoon — APPROVED** [review]

**Pull request merged**